doc/reference/drivers/index.rst

```diff
@@ -205,26 +205,66 @@ A device-specific API definition typically looks like this:
 
    #include <drivers/subsystem.h>
 
-   int specific_do_this(struct device *device, int foo);
+   /* When extensions need not be invoked from user mode threads */
+   int specific_do_that(struct device *device, int foo);
+
+   /* When extensions must be invokable from user mode threads */
+   __syscall int specific_from_user(struct device *device, int bar);
+
+   /* Only needed when extensions include syscalls */
+   #include <syscalls/specific.h>
 
 A driver implementing extensions to the subsystem will define the real
 implementation of both the subsystem API and the specific APIs:
 
 .. code-block:: C
 
-   static int generic_do_whatever(struct device *device, void *arg)
+   static int generic_do_this(struct device *device, void *arg)
+   {
+      ...
+   }
+
+   static struct generic_api api {
+      ...
+      .do_this = generic_do_this,
+      ...
+   };
+
+   /* supervisor-only API is globally visible */
+   int specific_do_that(struct device *device, int foo)
    {
       ...
    }
 
-   static int specific_do_this(struct device *device, int foo)
+   /* syscall API passes through a translation */
+   int z_impl_specific_from_user(struct device *device, int bar)
    {
       ...
    }
 
+   #ifdef CONFIG_USERSPACE
+
+   #include <syscall_handler.h>
+
+   int z_vrfy_specific_from_user(struct device *device, int bar)
+   {
+       Z_OOPS(Z_SYSCALL_SPECIFIC_DRIVER(dev, K_OBJ_DRIVER_GENERIC, specific_init));
```

Comment:
`Z_OOPS(Z_SYSCALL_SPECIFIC_DRIVER(dev, from_user))`

Reply:
No; the example was out of date because a subsequent change switched validation from the init function to the api pointer, but the suggested replacement is not correct.